src/domain/reports/mal-data-approval/usecases/UpdateMalApprovalStatusUseCase.ts

```diff
@@ -15,8 +25,36 @@ export class UpdateMalApprovalStatusUseCase {
                 return this.approvalRepository.complete(items);
             case "approve":
                 return this.approvalRepository.approve(items);
-            case "duplicate":
-                return this.approvalRepository.duplicateDataSets(items);
+            case "duplicate": {
+                const dataElementsWithValues: DataDiffItemIdentifier[] = _(
```

Comment:
can we extract this into a private method so we keep the code inside the case shorter.

```ts
const dataElementsWithValues = this.getDataElementsToDuplicate()
```

Feel free to use whatever name you consider is best for the method.